### DISCUSSION TITLE
Customising HR92 behaviour

### DISCUSSION THREAD

**Discussion by dsl101** [ghoti57/evofw3#44] [Q&A]
I was following with interest [this discussion](https://github.com/ghoti57/evofw3/issues/11) as I have a related issue with the HR92 TRVs, and was hoping someone might have some insight into whether what I'd like to do is possible.

The problem I experience is water hammer when the valves are almost closed; we've tried the usual remedies such as making sure the boiler internal bypass is at maximum, reducing the pump speed, and even fitted an external bypass valve—but even when this is as open as possible, there is still some hammer when only 1 or 2 rads are demanding heat, but are very close to their set temperature. The worst-case scenario is when one rad demands 100%, but another is at minimum; the noise is deafening throughout the house.

I'm convinced this is a design issue with the TRVs—they are bi-directional (a common cause of water hammer is fitting a uni-directional valve on the wrong side of the radiator) but just seem very prone to it. And here's my 'simple solution'... :)

I'm assuming somewhere in the firmware of the HR92 is a lookup table for demand vs valve position (there's no setting for anything like this in the default firmware). I'd like to tweak that table such that the valve cannot be set to the 'lowest' setting other than completely closed. This might result in some slight overshoot of temperature before the valve closes completely, but that's a very small price to pay for silence.

So, what are the chances of (a) getting the firmware off that device, (b) figuring out where the table is (if it even exists—of course it may just be a straight calculation?), and (c) getting a customised version back on? I do have a faulty HR92 (temperature sensors appears to have died) to experiment with, at least for getting firmware on / off. Any pointers for using JTAG or SPI with these devices—I'm a software engineer so relatively technical, but _not_ at this level; the last time I did any low-level stuff was at uni 30 years ago.

**Comment by Maikel-K** [comment]
I had the same, and fitted Heimeier Eclipse valve so that the flow stays the same.

This solved my issues.

**Reply by dsl101** [reply]
Thanks—that is an option, but we've just had 13 Honeywell valves fitted, and would like to avoid spending even more if poss :/ If the HR92 had a 'min flow' parameter, I feel that would solve it too, but I can't find anything like that.

**Reply by ghoti57** [reply]
It's unlikely you will find a way to update/modify the HR92 firmware.

It's also going to be virtually impossible to use RAMSES messages to override low demand settings.  The HR92 sleeps most of the time and only wakes just before the system controller is due to send an update (approx every 3.5 minutes, varies by controller).  It will go back to sleep almost immediately so the chance of recognizing the need for an override and getting it transmitted in time would be challenging.  If you don't succeed you'll just get the hammer.

Trying to use a proxy between the controller and HR92 s  means your heating will be 'off' if your proxy dies.

**Reply by dsl101** [reply]
Maybe I'm speculating on how the HR92 actually works, but I was assuming it used the local temperature measurement along with the target temperature to calculate the valve position—is that not how it works? And the update messages just sent a new target temperature rather than the demand?

But even if the messages do send the desired demand, again I was assuming this would be a percentage or similar, the HR92 would then convert that into a valve position, and it was that step I was hoping to change. So that, say 5% demand would open the valve no less than 10% demand (if that's where the hammering stops), and then 0% would close the valve fully.
